services/cd-service/pkg/repository/repository.go

```diff
@@ -369,10 +370,21 @@ func (r *repository) Apply(ctx context.Context, transformers ...Transformer) err
 }
 
 func (r *repository) Push(ctx context.Context, pushAction func() error) error {
+
+	span, ok := tracer.SpanFromContext(ctx)
```

Comment:
I think `StartSpanFromContext` is actually the better option here, as it creates a new Span, if none exists.

Reply:
```suggestion
	span, ctx := tracer.StartSpanFromContext(ctx, "Push")
```
Note that it returns a context which we need to use in the rest of the function